newsroom/agenda/agenda.py

```diff
@@ -318,10 +313,19 @@ def _agenda_query():
 def get_date_filters(args):
     date_range = {}
     offset = int(args.get("timezone_offset", "0"))
+
+    agenda_filters = app.config.get("AGENDA_TIME_FILTERS")
+    default_filter = next((f for f in agenda_filters if f.get("default", False)), None)
+
     if args.get("date_from"):
-        date_range["gt"] = get_local_date(args["date_from"], "00:00:00", offset)
+        if default_filter and not args.get("date_to") and args.get("date_from") == get_utcnow().strftime("%Y-%m-%d"):
+            date_range["gt"] = get_local_date(default_filter["query"], "00:00:00", offset)
```

Comment:
it seems like those filters are always limiting the upper bound not the lower one, so to make it work properly we should probably send that as a `date_to` or a new `date_filter` option and work with that

Reply:
actually thinking about it the filtering for those options should filter events which
end after the given filter, so if the filter is next 7 days the filter should be:
```
date_range["lt"] = default_filter["query"]
```
which will later use a filter like:
```
dates.end <= date_range["lt"]
```
and you can keep the `date_range["gt"]` for selected day, that should still work the same probably.